File: tests/fixtures/sample.java
// Java Example - Annotation & Generic Focus
import java.util.List;
import java.util.ArrayList;

@Service  // Annotation → META layer
public class UserService {
    private final Database db;  // Final → readonly
    
    @Inject  // Dependency injection annotation
    public UserService(Database db) {
        this.db = db;
    }
    
    @Override  // Override annotation → DOCUMENTATION
    public String toString() {
        return "UserService{db=" + db + "}";
    }
    
    // Generics → DECLARATION layer
    public <T extends Comparable<T>> List<T> sortItems(List<T> items) {
        var result = new ArrayList<>(items);  // var → type inference
        result.sort(Comparable::compareTo);   // Method reference → CONTROL FLOW
        return result;
    }
    
    // Lambda expression → CONTROL FLOW
    public void processUsers(List<User> users) {
        users.forEach(user -> {
            if (user.isActive()) {
                user.save();  // Mutation
            }
        });
    }
    
    // Synchronized → CONTROL FLOW
    public synchronized void incrementCounter() {
        counter++;  // Mutation under lock
    }
    
    // Try-catch → CONTROL FLOW
    public void riskyOperation() throws IOException {
        try {
            db.connect();
        } catch (SQLException e) {
            throw new IOException("Connection failed", e);
        } finally {
            db.close();
        }
    }
    
    // Static access → USAGE with static modifier
    public static final int MAX_USERS = Integer.MAX_VALUE;
}

// Record (Java 14+) → DECLARATION
record Point(int x, int y) {}

// Sealed class (Java 17+) → META
sealed class Shape permits Circle, Rectangle {}

File: tests/fixtures/sample.py
# Python Example - Declaration Focus
from dataclasses import dataclass
from typing import List

@dataclass
class User:
    """User data model with intent-based coloring."""
    name: str
    email: str
    active: bool = True
    
    async def save_to_database(self, db: Database) -> None:
        """Save user to database with async/await."""
        # Mutation: state changes
        self.active = True
        await db.save(self)
        
    @property
    def display_name(self) -> str:
        """Property decorator → META layer."""
        return f"{self.name} <{self.email}>"

async def fetch_users(limit: int = 10) -> List[User]:
    """Async function → CONTROL FLOW + async modifier."""
    users = []
    
    # List comprehension → DATA layer
    active_users = [u for u in users if u.active]
    
    # Control flow: if statement
    if len(active_users) > limit:
        return active_users[:limit]
    
    return active_users

# Global scope modifier → MUTATION
global user_cache
user_cache = {}
